src/transformers/trainer.py

```diff
@@ -1977,6 +1987,13 @@ def zero_grad(x):
                 if self.control.should_epoch_stop or self.control.should_training_stop:
                     break
 
+            if self.args.xla_measure_avg_step_time:
+                num_step = len(epoch_iterator)-xla_avg_step_time_starting_step
+                xla_avg_step_time = (time.time()-xla_avg_step_time_begin_time-xla_tracing_time)/num_step
```

Comment:
You may need one more `wait_device_ops` here before calculating the total time

Also an additional `mark_step` since I think our branch is behind master: https://github.com/huggingface/transformers/blob/a0e77a1f6bdfcceccdc5618e8a01ee32ef47bfa8/src/transformers/trainer.py#L2286-L2290

Reply:
Good catch!

Reply:
@vanbasten23 Do you mind rebase the branch to the latest?

Reply:
Sure, Just checked origin:llama2-google-next-training-xiowei-measure-avg-step-time is update to date with upstream:llama2-google-next-training

Reply:
You can merge this branch. I mean rebasing llama2-google-next-training with hf main.